src/Dapr.Client/PublishHttpClient.cs

```diff
@@ -0,0 +1,95 @@
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Dapr.DaprUris;
+
+    /// <summary>
+    /// A client for interacting with the Dapr publish endpoints using <see cref="HttpClient" />.
+    /// </summary>
+    public class PublishHttpClient : PublishClient
+    {
+        private readonly HttpClient client;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PublishHttpClient"/> class.
+        /// </summary>
+        /// <param name="client">The <see cref="HttpClient" />.</param>
+        /// <param name="serializerOptions">The <see cref="JsonSerializerOptions" />.</param>
+        public PublishHttpClient(HttpClient client, JsonSerializerOptions serializerOptions = null)
+        {
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            this.client = client;
+            this.serializerOptions = serializerOptions;
+        }
+
+        /// <inheritdoc/>
+        public override async Task PublishEventAsync<TRequest>(string topicName, TRequest publishContent, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(topicName));
+            }
+
+            if (publishContent is null)
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(publishContent));
+            }
+
+            await this.MakePublishHttpRequest(topicName, JsonSerializer.Serialize(publishContent), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public override async Task PublishEventAsync(string topicName, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(topicName));
+            }
+
+            await this.MakePublishHttpRequest(topicName, string.Empty, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<HttpResponseMessage> MakePublishHttpRequest(string topicName, string publishContents, CancellationToken cancellationToken)
+        {
+            var url = this.client.BaseAddress == null ? $"http://localhost:{DefaultHttpPort}{PublishPath}/{topicName}" : $"{PublishPath}/{topicName}";
+            var request = new HttpRequestMessage(HttpMethod.Post, url);
+
+            request.Content = new StringContent(publishContents);
+
+            var response = await this.client.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                return default;
+            }
```

Comment:
The spec for pub/sub doesn't say anything about 404 being special, this should probably be removed.

https://github.com/dapr/docs/blob/master/reference/api/pubsub.md#pub-sub-and-broadcast